machine.go

```diff
@@ -634,12 +635,28 @@ func (m *Machine) Run(command string) (int, error) {
 	return m.startup(command, nil)
 }
 
+// Escape patterns \.+*?()|[]{}^$ with backslash
+func escapePatterns(s string, patterns []string) string {
```

Comment:
Thanks for your feedback.

>This comment seems to indicate that this should escape a whole list of patterns; But actually you pass the pattern and in >the only place that call it it just passes $ ?

I would like to provide an API that support for shell escaping with backslash.
In our case, we have error with dollar sign. 

>Would it make more sense to use some go shellescape pkg for this instead?

Do you mean https://pkg.go.dev/github.com/alessio/shellescape? It using single quote to escape shell.  I've tried it but it is not effective in case of  dollar sign with number (likes '$6').

Reply:
Hi @sjoerdsimons, that comment ("Escape patterns \.+*?()|[]{}^$ with backslash") is misleading/wrong. He is just escaping the dollar sign "$" and not escaping anything else.

The use case is he wants to pass an encrypted password to a debos recipe and add it to /etc/shadow. For example something like this (in practice passed through the environment as "-t user_password_encrypted : $6$j6e3P.isi89LHTm$F...")

`{{- $user_password_encrypted := or .user_password_encrypted "$6$j6e3P.isi89LHTm$F8Dtg./a245AtyUQkBcEeth/ArvZ8pWs/RcKfJDc5UlAPl3dnurTPEyshDNIpAPfe.XR0NwoQZa.m4YOV/4uo1" -}}`

According to https://linux.die.net/man/3/crypt, encrypted passwords have characters chosen from the set [a-zA-Z0-9./] plus the $ for separating the id and salt.

The problem he found is that "$" gets interpreted and by the time it arrives at the actual action the value is distorted.
```
  - action: run
    chroot: true
    script: setpassword.h --encrypted-password '{{ $skelios_user_password_encrypted }}'
```

Having said that, that is our use case.
It seems to also solve this user's use case as well: https://github.com/go-debos/debos/issues/141

I am not sure this covers all use cases and therefore it might be just a bad workaround.
But the user of the recipe's interface expects that the value is passed as it is, so this is a problem/bug that needs to be fixed.

Thanks,
Daniel

Reply:
Actually, we have tried using this patch for solving the issue https://github.com/go-debos/fakemachine/issues/32 but it doesn't work. 

In that issue, fakemachine is called directly as al command (i.e. not from debos), so he needs to add shell escaping to the command as follows:
```
fakemachine -- python -c "'import sys;print(sys.argv)'" "\"foo bar"\"
```

Reply:
For just replacing the "$" the code is a bit convoluted. He only needs to call strings.Replace(...) I think. So if you are interested he can resend. Having said that, I wonder if this is the only use case. I guess once other users have similar problems we can come with a more generic solution